swagger.yaml

```diff
@@ -1,42 +1,390 @@
+      offset:
+        description: "The first row of images to retrieve, starting at 0. Use this parameter as a pagination mechanism along with the limit parameter"
+        type: integer
+      total_count:
+        description: "The total number of images"
+        readOnly: true
+        type: integer
+        example: 1
+
+  NewImage:
+    type: object
+    description: "New Image metadata model, used to create new images. The image ID will be created by image API, and the state will be set to 'created' at creation time. If 'category' is provided but 'download' variants are not provided, then the default download variants for the provided category will be generated."
+    properties:
+      collection_id:
+        type: string
+        description: "Collection unique identifier corresponding to this image"
+        example: "5557dcd9-bf58-4a67-94f7-2343569834cc"
+      filename:
+        type: string
+        description: "Image's file name. According to SEO recommendations, the name should not be longer than 5 words. And it should not include extension because multiple extensions for the same image might be available as download variants."
+        example: "image-name"
+        maxLength: 40
+      license:
+        type: object
+        description: "License that defines the third party permissions for this image"
+        properties:
+          title:
+            type: string
+            description: "Title of the license"
+            example: "Open Government Licence v3.0"
+          href: 
+            type: string
+            description: "Link to the license content"
+            example: "https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+      upload:
+        type: object
+        description: "Original file upload in the upload bucket"
+        properties:
+          path:
+            type: string
+            description: "S3 object key (without bucket name) where the originally uploaded image is stored"
+            example: "images/025a789c-533f-4ecf-a83b-65412b96b2b7/image-name.png"
+      category:
+        type: string
+        description: "Category of image, which might define a set of possible formats and variants or resolutions."
+        example: "chart"
+      downloads:
+        type: object
+        description: "All available formats and variants for this image. Note that this spec does not define all cases, and depending on the image category, we can have multiple formats as well as multiple variants."
+        properties:
+          png:
+            type: object
+            description: "All available download variants for this image format, keyed by resolution (in WxH pixels) or with some other identifier."
+            properties:
+              1920x1080:
+                $ref: '#/definitions/ImageDownload'
+              1280x720:
+                $ref: '#/definitions/ImageDownload'
+              thumbnail:
+                $ref: '#/definitions/ImageDownload'      
```

Comment:
The image downloads will be created by the importer. At image creation stage, only the upload is known. So I don't think this entire section is needed in this definition.

Reply:
thanks, I will remove it from there and fix the description